File: src/main/java/by/ladyka/club/entity/Event.java
package by.ladyka.club.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

import static by.ladyka.club.ClubApplication.APP_TABLE_PREFIX;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = APP_TABLE_PREFIX + "event")
@EntityListeners(AuditingEntityListener.class)
public class Event extends AbstractEntity{
    public static final String startEventFieldName = "startEvent";
    private BigDecimal costMinimum;
    @Lob
    private String costText;
    @Lob
    private String description;
    private LocalDateTime startEvent;
    private LocalDateTime endEvent;
    private String name;
    private String coverUri;
    private int status;
    private String alias;

    public Event(Long eventId) {
        setId(eventId);
    }
}

File: src/main/angular/src/app/dto/event.ts
export class Event {
  id: number;
  cost: number;
  costText: string;
  description: string;
  startEvent: string;
  endEvent: string;
  name: string;
  coverUri: string;
  buyTicketUrl: string;
  recommendation: boolean;
  republicPay: boolean;
  costDance: number;
  costTablePlace: number;
}

export class EventListResult {
  items: Event[];
  total: number
}
